src/dbos-executor.ts

```diff
@@ -1921,19 +1886,27 @@ export class DBOSExecutor implements DBOSExecutorContext {
     }
   }
 
-  async getWorkflowStatus(workflowID: string, callerID?: string, callerFN?: number): Promise<WorkflowStatus | null> {
+  async getWorkflowStatus(
+    workflowID: string,
+    callerID?: string,
+    callerFN?: number,
+    getRequest: boolean = false,
+  ): Promise<WorkflowStatus | null> {
+    // use sysdb getWorkflowStatus directly in order to support caller ID/FN params
     const status = await this.systemDatabase.getWorkflowStatus(workflowID, callerID, callerFN);
-    return status ? DBOSExecutor.toWorkflowStatus(status, true) : null;
+    return status ? toWorkflowStatus(status, getRequest) : null;
   }
 
-  async listWorkflows(input: GetWorkflowsInput): Promise<WorkflowStatus[]> {
-    const wfs = await this.systemDatabase.listWorkflows(input);
-    return wfs.map((wf) => DBOSExecutor.toWorkflowStatus(wf, true));
+  async listWorkflows(input: GetWorkflowsInput, getRequest: boolean = false): Promise<WorkflowStatus[]> {
```

Comment:
There's probably something I'm missing, but do these pass-through methods need to exist? Or can their users call the API in `dbos.ts` directly?

Reply:
SystemDatabase works in terms of `WorkflowStatusInternal`. Public APIs expose `WorkflowStatus`. Executor and Client both call the helper functions in `workflow-management.ts`